config/defaults.mk

```diff
@@ -394,6 +395,11 @@ AMGX_DIR = @MFEM_DIR@/../amgx
 AMGX_OPT = -I$(AMGX_DIR)/include
 AMGX_LIB = -L$(AMGX_DIR)/lib -lamgx -lcusparse -lcusolver -lcublas -lnvToolsExt
 
+# MAGMA library configuration
+MAGMA_DIR = @MFEM_DIR@/../magma
+MAGMA_OPT = -I$(MAGMA_DIR)/include
+MAGMA_LIB = -L$(MAGMA_DIR)/lib -l:libmagma.a -lcublas -lcusparse $(LAPACK_LIB)
```

Comment:
Regarding `-l:libmagma.a` -- do we really want this as the default? I suspect `-lmagma` is a better default.

Regarding `-lcublas -lcusparse` -- to allow users to switch easily to HIP, I suggest we add a separate variable, e.g. `MAGMA_GPU_LIB` which be default is set for CUDA and the setting for HIP is provided as a comment:
```make
MAGMA_GPU_LIB = -lcublas -lcusparse
# MAGMA_GPU_LIB = <required hip libraries>
```

Right now, MFEM CUDA build has cuSparse as a dependency and this PR is adding cuBLAS as a dependency, so maybe `MAGMA_GPU_LIB` is not strictly necessary but things may change in the future.

Reply:
MAGMA by default builds both a shared library and a static library. `-lmagma` was finding the shared library, which for some reason wasn't working for me (I don't remember exactly what went wrong).

For `-lcublas -lcusparse`, maybe we can just let `CUDA_LIB` and `HIP_LIB` take care of these.

Reply:
> MAGMA by default builds both a shared library and a static library. `-lmagma` was finding the shared library, which for some reason wasn't working for me (I don't remember exactly what went wrong).

If there was an issue using the shared library, then it is better to leave the explicit linking with the static one.

> For `-lcublas -lcusparse`, maybe we can just let `CUDA_LIB` and `HIP_LIB` take care of these.

Up to you, if it works (removing the gpu-related flags `-lcublas -lcusparse`), we can use it. If things change in the future we can update the defaults.